lms/djangoapps/save_for_later/management/commands/send_course_reminder_emails.py

```diff
@@ -0,0 +1,88 @@
+
+
+class Command(BaseCommand):
+    """
+    Command to send reminder emails to those users who
+    saved course by email but not register within 15 days.
+
+
+    Examples:
+
+        ./manage.py lms send_course_reminder_emails --batch-size=100
+    """
+    help = dedent(__doc__)
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--batch-size',
+            type=int,
+            default=1000,
+            help='Maximum number of users to send reminder email in one chunk')
+
+    def handle(self, *args, **options):
+        """
+        Handle the send save for later reminder emails.
+        """
+
+        reminder_email_threshold_date = datetime.now() - timedelta(
+            days=settings.SAVE_FOR_LATER_REMINDER_EMAIL_THRESHOLD)
+        saved_courses_ids = SavedCourse.objects.\
+            filter(reminder_email_sent=False, modified__lt=reminder_email_threshold_date).values_list('id', flat=True)
+        total = saved_courses_ids.count()
+        batch_size = max(1, options.get('batch_size'))
+        num_batches = ((total - 1) / batch_size + 1) if total > 0 else 0
+
+        for batch_num in range(int(num_batches)):
+            start = batch_num * batch_size
+            end = min(start + batch_size, total) - 1
+            saved_courses_batch_ids = list(saved_courses_ids)[start:end + 1]
+
+            query = SavedCourse.objects.\
+                filter(id__in=saved_courses_batch_ids).order_by('-modified')
+            saved_courses = use_read_replica_if_available(query)
+            for saved_course in saved_courses:
+                user = None
+                if saved_course.user_id:
+                    user = User.objects.filter(id=saved_course.user_id).first()
+                course_overview = CourseOverview.get_from_id(saved_course.course_id)
+                course_data = {
+                    'course': course_overview,
+                    'send_to_self': None,
+                    'user_id': saved_course.user_id,
+                    'org_img_url': saved_course.org_img_url,
+                    'marketing_url': saved_course.marketing_url,
+                    'weeks_to_complete': saved_course.weeks_to_complete,
+                    'min_effort': saved_course.min_effort,
+                    'max_effort': saved_course.max_effort,
+                    'type': 'course',
+                    'reminder': True,
+                }
+                if user:
+                    enrollment = CourseEnrollment.get_enrollment(user, saved_course.course_id)
+                    if enrollment:
+                        return
+                email_sent = send_email(saved_course.email, course_data)
+                if not email_sent:
+                    logging.info("Unable to send reminder email to {user} for {course} course"
+                                 .format(user=str(saved_course.user_id), course=str(saved_course.course_id)))
+            saved_courses.update(reminder_email_sent=True)
```

Comment:
What if for some courses the email sending was failed, I believe we shouldn't set this to true for them. Instead of bulk updating, we should update them individually depending upon the email sent status.

Reply:
One option could be to maintain a list of record ids for which the email was failed and exclude them while bulk updating.